eng/scripts/Cadl-Project-Sync.ps1

```diff
@@ -0,0 +1,101 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+
+function AddSparseCheckoutPath([string]$subDirectory, [string]$file) {
+    if (!(Test-Path $file) -or !((Get-Content $file).Contains($subDirectory))) {
+        Write-Output $subDirectory >> .git/info/sparse-checkout
+    }
+}
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
+
+    $currentDur = Resolve-Path "."
+    Write-Host "Generating from $currentDur"
+    if (Test-Path ".\package.json") {
+        Remove-Item -Path ".\package.json" -Force
+    }
+    if (Test-Path ".\.npmrc") {
+        Remove-Item -Path ".\.npmrc" -Force
+    }
+    npm install "@azure-tools/cadl-csharp"
+
+    Pop-Location
+}
+
+function CopySpecToProjectIfNeeded([string]$pullResult, [string]$specCloneRoot, [string]$mainSpecDir, [string]$dest, [string[]]$specAdditionalSubDirectories) {
+    if (!($pullResult.Contains("Already up to date.")) && Test-Path $dest) {
+
+        $source = "$specCloneRoot/$mainSpecDir"
+        Write-Host "Copying spec from $source"
+        Copy-Item -Path $source -Destination $dest -Recurse -Force
+        foreach($additionalDir in $specAdditionalSubDirectories)
+        {
+            $source = "$specCloneRoot/$additionalDir"
+            Write-Host "Copying spec from $source"
+            Copy-Item -Path $source -Destination $dest -Recurse -Force
+        }
+    }
+}
+
+function UpdateSparseCheckoutFile([string]$mainSpecDir, [string[]]$specAdditionalSubDirectories) {
+    $sparseCheckoutFile = ".git/info/sparse-checkout"
+    AddSparseCheckoutPath $mainSpecDir $sparseCheckoutFile
+    foreach($subDir in $specAdditionalSubDirectories)
+    {
+        AddSparseCheckoutPath $subDir $sparseCheckoutFile
+    }
+}
+
+function InitializeSparseGitClone([string]$repo) {
+    if (!(Test-Path ".git")) {
+        git init
+        git remote add origin $repo
+        git config core.sparseCheckout true
```

Comment:
```suggestion
        git init
        if ($LASTEXITCODE) { exit $LASTEXITCODE }
        git remote add origin $repo
        if ($LASTEXITCODE) { exit $LASTEXITCODE }
        git config core.sparseCheckout true
        if ($LASTEXITCODE) { exit $LASTEXITCODE }
```

Reply:
For now, yes. Just an FYI, though: this is now automatic in PowerShell 7.3: "native" commands still set `$LASTEXITCODE` but that still works like `$?` and friends now, so the `$ErrorActionPreference` should work regardless. Again, though, for now this is a good suggestion.